code/datums/components/connect_containers.dm

```diff
@@ -0,0 +1,62 @@
+/// This component behaves similar to connect_loc_behalf, but it's nested and hooks a signal onto all MOVABLES containing this atom.
+/datum/component/connect_containers
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+
+	/// An assoc list of signal -> procpath to register to the loc this object is on.
+	var/list/connections
+	var/atom/movable/tracked
+
+/datum/component/connect_containers/Initialize(atom/movable/tracked, list/connections)
+	. = ..()
+	if (!ismovable(tracked))
+		return COMPONENT_INCOMPATIBLE
+
+	src.connections = connections
+	src.tracked = tracked
+
+/datum/component/connect_containers/InheritComponent(datum/component/component, original, atom/movable/tracked, list/connections)
+	// Not equivalent. Checks if they are not the same list via shallow comparison.
+	if(!compare_list(src.connections, connections))
+		return
+	if(src.tracked != tracked)
+		set_tracked(tracked)
+
+/datum/component/connect_containers/RegisterWithParent()
+	set_tracked(tracked, TRUE)
+
+/datum/component/connect_containers/UnregisterFromParent()
+	set_tracked(null)
+
+/datum/component/connect_containers/proc/set_tracked(atom/movable/new_tracked, first_run = FALSE)
+	if(tracked && !first_run)
+		UnregisterSignal(tracked, COMSIG_MOVABLE_MOVED)
+		unregister_signals(tracked.loc)
+	tracked = new_tracked
+	if(!tracked)
+		return
+	RegisterSignal(tracked, COMSIG_MOVABLE_MOVED, .proc/on_moved)
+	update_signals(tracked)
+
```

Comment:
Why does first_run exist?

Reply:
Bad code. RegisterWithParent is called after Initialize (which sets the tracked variable), so that the statement on line 31 could be reached, even if we know we have no signals to unregister from tracked or anything else yet.
I just removed the RegisterWithParent and UnregisterWithParent definitions here because they're only relevant for transferable components and moved the set_tracked() call to Initialize so the first_run arg isn't needed anymore.